scripts/Dockerfile

```
@@ -21,45 +21,44 @@ ENV CMAKE_C_COMPILER=/usr/bin/gcc
 ENV CMAKE_CXX_COMPILER=/usr/bin/g++
 ENV CMAKE_MAKE_PROGRAM=/usr/bin/make
 ENV SUNDIALS_INST=/home/pybamm/.local
-ENV LD_LIBRARY_PATH=/home/pybamm/.local/lib:
+ENV LD_LIBRARY_PATH=/home/pybamm/.local/lib
+
+RUN conda create -n pybamm python=3.11
+RUN conda init --all
+SHELL ["conda", "run", "-n", "pybamm", "/bin/bash", "-c"]
+RUN conda install -y pip
 
 ARG IDAKLU
 ARG ODES
 ARG JAX
 ARG ALL
 
-RUN conda create -n pybamm python=3.9
-RUN conda init --all
-SHELL ["conda", "run", "-n", "pybamm", "/bin/bash", "-c"]
-RUN conda install -y pip
+RUN pip install --upgrade --user pip setuptools wheel wget nox cmake
```

Hmm, `nox` should be installed as a dev dependency -

https://github.com/pybamm-team/PyBaMM/blob/ec5383c7538ed54d41ea1afbbf61f076992535e1/setup.py#L256

```suggestion
RUN pip install --upgrade --user pip setuptools wheel wget cmake
```